tests/api.rs

```diff
@@ -0,0 +1,83 @@
+mod support;
+
+#[cfg(feature = "api")]
+mod tests {
+    use crate::support::{sink, source};
+    use reqwest::Response;
+    use std::net::SocketAddr;
+    use std::time::Duration;
+    use vector::api;
+    use vector::config::Config;
+    use vector::test_util::{next_addr, retry_until};
+
+    // provides a config that enables the API server, assigned to a random port. Implicitly
+    // tests that the config shape matches expectations
+    fn api_enabled_config() -> Config {
+        let mut config = Config::builder();
+        config.add_source("in1", source().1);
+        config.add_sink("out1", &["in1"], sink(10).1);
+        config.api.enabled = true;
+        config.api.bind = Some(next_addr());
```

Comment:
`next_addr()` assumes port 1234 is available. Ran into a situation where I had a running instance of [Parcel](https://parceljs.org/), which uses this same port by default. Although this is probably uncommon, it led to some head stratching.

I think it's worth switching this to a real, free port -- will tackle in a separate PR.

Reply:
You can have the OS assign you a free port by asking for port `0`.

Reply:
That's actually what portpicker does (after attempting a random port first). It momentarily spawns a TCP listener, copies the port, and then drops it.

Reply:
Cool!